### PR TITLE
allow custom Extrinsic implementation

### PR DESCRIPTION
**Pull request by ianhe8x**
see #738
Originally part of refactor for Metadata V2.

### CODE REVIEW

File: packages/api/src/SubmittableExtrinsic.ts [hunk 1]
```diff
@@ -3,8 +3,9 @@
 // of the Apache-2.0 license. See the LICENSE file for details.
 
 import { KeyringPair } from '@polkadot/keyring/types';
-import { AccountId, Address, Extrinsic, ExtrinsicStatus, EventRecord, Hash, Index, Method, SignedBlock, Struct, Text, Vector } from '@polkadot/types/index';
-import { Codec, CodecCallback, SignatureOptions } from '@polkadot/types/types';
+import { AccountId, Address, ExtrinsicStatus, EventRecord, Hash, Index, Method, SignedBlock, Struct, Text, Vector } from '@polkadot/types/index';
+import { Codec, CodecCallback, ExtrinsicLike, SignatureOptions } from '@polkadot/types/types';
+import getRegistry from '@polkadot/types/codec/typeRegistry';
```
Comment (jacogr) [inline packages/api/src/SubmittableExtrinsic.ts]: Just export `getRegistry` in `@polkadot/codec/index` (so it is available everywhere as part of `@polkadot/types`)

**Comment by ianhe8x** [comment]
The ExtrinsicInterface is the place i'm not quite certain at the moment.` { hash: Hash }` is fine (i guess no parachain would go into that deep to replace Hash with other implementation), but as to `signature` I'm not sure if Codec is too generic.

**Comment by jacogr** [comment]
For `Signature`, I would go for U8a as well, seems reasonable and the lowest-level. (Since it probably should end up with an array)

We probably need to have an `IHash` and `ISignature` for these.

**Comment by jacogr** [comment]
"try to make codeclimate happy" - an impossible task. 

I'll give this a look-over and play with the e2e tests as well.

**Comment by jacogr** [comment]
Looks good code wise. Will just do the e2e tests in the morning. (Sorry for the slight delay on my side getting this in)

**Comment by ianhe8x** [comment]
`makes a transfer, swaps types and then another one` in the e2e test fails, though I tried master branch that testcase fails as well, so i guess that's fine.

**Comment by jacogr** [comment]
That one is “fine” since sr25519 is still unused and there is some ongoing work looking at key derivation, however something broke in keyring so needs a fix soon-ish. (However completely out of scope here)

**Pull request merged**